src/ring_theory/polynomial/basic.lean

```diff
@@ -96,6 +96,42 @@ begin
   exact lt_of_le_of_lt (degree_X_pow_le _) (with_bot.coe_lt_coe.2 $ finset.mem_range.1 hk)
 end
 
+/-- Coefficient linear map on degree_lt -/
+def degree_lt_linear_map (F : Type*) [field F] (n : ℕ) :
```

Comment:
I just realized that we have `linear_algebra/lagrange.lean`. I think there is overlap. Can you reuse/adapt/modify what is in that file?

Reply:
The equivalence shown in lagrange.lean requires picking distinct elements of the field, which might not be possible if the field is finite.

Reply:
I don't think that there's any overlap between the two approaches to proving the equivalence

Reply:
Can we find a common generalization? Using multisets? Or elements indexed by a fintype?

Reply:
Hmmm, ok. I would like to think a bit about this. At least, If we keep this approach, I think it should be flagged as a variant on lagrange.

Reply:
It is true that we could prove a generalization of the form
`def fun_equiv_degree_lt (a : Type*) (s : fintype a) : degree_lt F s.card ≃ₗ[F] ((↑s : set a) → F) :=`
Then the lagrange version is the special case where `s : finset F` and the polynomial version is the special case where `s = (finset.range n) : finset ℕ`

Reply:
Hmm, on second thought I think that they do serve different purposes. But I would like to change the names and docs a bit in this PR.

Reply:
@tb65536 I've pushed a golfed version. It is also stated in terms of `fin n` instead of casting `finset.range n` to a type. I hope that doesn't cause trouble for the applications in Galois theory.